assets/app/scripts/directives/nav.js

```diff
@@ -20,86 +20,78 @@ angular.module('openshiftConsole')
       templateUrl: "views/_sidebar-main-nav-item.html"
     };
   })
-  .directive('projectNav', function($timeout, $location, $filter, LabelFilter) {
+  .directive('projectNav', function($timeout, $location, $filter, LabelFilter, DataService) {
     return {
-      restrict: 'E',
+      restrict: 'EA',
       templateUrl: 'views/_project-nav.html',
-      link: function ($scope, element) {
-        var select = $('.selectpicker', element);
+      link: function($scope, $elem) {
+        var select = $elem.find('.selectpicker');
+        var projects = {};
+        var sortedProjects = [];
+        var options = [];
```

Comment:
instead of this all being in link, should we be populating options from a controller on the directive instead, and making the data request from the controller as well.  keep the link function down to the things that have to happen after the template compilation occurs.